ambry-store/src/main/java/com.github.ambry.store/DiskManager.java

```diff
@@ -0,0 +1,163 @@
+  private final DiskId disk;
+  private final StoreConfig config;
+
+  private static final Logger logger = LoggerFactory.getLogger(DiskManager.class);
+
+  /**
+   * Constructs a {@link DiskManager}
+   * @param disk representation of the disk.
+   * @param replicas all the replicas on this disk.
+   * @param config the settings for store configuration.
+   * @param scheduler the {@link ScheduledExecutorService} for executing background tasks.
+   * @param registry the {@link MetricRegistry} used for store-related metrics.
+   * @param keyFactory the {@link StoreKeyFactory} for parsing store keys.
+   * @param recovery the {@link MessageStoreRecovery} instance to use.
+   * @param hardDelete the {@link MessageStoreHardDelete} instance to use.
+   * @param time the {@link Time} instance to use.
+   */
+  DiskManager(DiskId disk, List<ReplicaId> replicas, StoreConfig config, ScheduledExecutorService scheduler,
+      MetricRegistry registry, StoreKeyFactory keyFactory, MessageStoreRecovery recovery, MessageStoreHardDelete hardDelete, Time time) {
+    this.disk = disk;
+    this.config = config;
+    DiskIOScheduler ioScheduler = new DiskIOScheduler(null);
+    for (ReplicaId replica : replicas) {
+      if (disk.equals(replica.getDiskId())) {
+        String storeId = replica.getPartitionId().toString();
+        BlobStore store = new BlobStore(storeId, config, scheduler, ioScheduler, registry, replica.getReplicaPath(),
+            replica.getCapacityInBytes(), keyFactory, recovery, hardDelete, time);
+        stores.put(replica.getPartitionId(), store);
+      }
+    }
+  }
+
+  /**
+   * Starts all the stores on this disk.
+   */
+  void start()
+      throws StoreException {
+    File mountPath = new File(disk.getMountPath());
+    if (mountPath.exists()) {
+      ScheduledExecutorService storeStartupScheduler = Utils
+          .newScheduler(Math.min(stores.size(), config.storeNumStoreStartupThreads), "store-startup-" + disk, false);
+      try {
+        Map<PartitionId, Future<Void>> futures = new HashMap<>();
+        for (final Map.Entry<PartitionId, BlobStore> partitionAndStore : stores.entrySet()) {
+          Future<Void> future = storeStartupScheduler.submit(new Callable<Void>() {
+            @Override
+            public Void call()
+                throws Exception {
+              partitionAndStore.getValue().start();
+              return null;
+            }
+          });
+          futures.put(partitionAndStore.getKey(), future);
+        }
+        int numFailed = 0;
+        for (Map.Entry<PartitionId, Future<Void>> partitionAndFuture : futures.entrySet()) {
+          try {
+            partitionAndFuture.getValue().get();
+          } catch (ExecutionException e) {
+            numFailed++;
+            logger.error("Exception while starting store for partition " + partitionAndFuture.getKey(), e.getCause());
+          } catch (InterruptedException e) {
```

Comment:
for simplicity, I think you should just catch `Exception` (or do `catch (ExecutionException | InterruptedException e)`) and do `logger.error("Exception while starting store", e)`. I don't think modifying the error message has any value since the exception gets printed.